test/unit/init/cluster/SingleThreaded.test.ts

```diff
@@ -0,0 +1,86 @@
+import { ComponentsManager } from 'componentsjs';
+import type { Resource } from '../../../../src';
+import { errorOnSingleThreadedComponents } from '../../../../src';
+
+const mainModulePath = 'C:\\Users\\jdoe\\Workspaces\\CommunitySolidServer';
+const mockComponentsManager = jest.fn().mockImplementation((nr: number): any => {
```

Comment:
Is there any way to make the tests work without all these mocks? Or is that unfeasible due to how components.js works?

Reply:
It's hard, feel free to try if you can reduce the mocking, but I couldn't get it to work otherwise. I remove some unnecessary arguments.

Reply:
Had a look, actually there is less mocking than I first thought. Most if it is the `mockComponentsManager` which is just a large function but whose behaviour is not really dependent on Components.js. So perhaps extract that as a separate function and fixed `moduleState` object so the actual mock is smaller?

Is the `...jest.requireActual('jsonld-context-parser')` necessary? I would expect only Components.js to depend on that, and that one is also being mocked.

Reply:
> Had a look, actually there is less mocking than I first thought. Most if it is the `mockComponentsManager` which is just a large function but whose behaviour is not really dependent on Components.js. So perhaps extract that as a separate function and fixed `moduleState` object so the actual mock is smaller?

Nice, this works.

> Is the `...jest.requireActual('jsonld-context-parser')` necessary? I would expect only Components.js to depend on that, and that one is also being mocked.

Yes, without it, I get this:
```
    TypeError: Cannot read properties of undefined (reading 'INVALID_INDEX_VALUE')

      at Object.<anonymous> (node_modules/jsonld-streaming-parser/lib/entryhandler/keyword/EntryHandlerKeywordUnknownFallback.js:47:80)
      at Object.<anonymous> (node_modules/jsonld-streaming-parser/lib/JsonLdParser.js:18:46)
```
INVALID_INDEX_VALUE is an internal jsonld-context-parser thing..

Reply:
Solved it by doing this:
```ts
jest.mock('jsonld-context-parser/lib/ContextParser', (): any => ({
  // eslint-disable-next-line @typescript-eslint/naming-convention
  ContextParser: jest.fn().mockImplementation((): any => ({
    parse: jest.fn(async(): Promise<any> => ({
      expandTerm: jest.fn((): any => myExpandTerm()),
    })),
  })),
}));
```
instead of this:
```ts
jest.mock('jsonld-context-parser', (): any => ({
   ...jest.requireActual('jsonld-context-parser'),
  // eslint-disable-next-line @typescript-eslint/naming-convention
  ContextParser: jest.fn().mockImplementation((): any => ({
    parse: jest.fn(async(): Promise<any> => ({
      expandTerm: jest.fn((): any => myExpandTerm()),
    })),
  })),
}));
```